### PR TITLE
Map ITRS frames to terrestrial WCS coordinates

### PR DESCRIPTION
**Pull request by lpsinger**
This will make it possible to use WCSAxes to make figures that combine both celestial and terrestrial features. An example is plotting the coordinates of an astronomical transient over an all- sky satellite image to illustrate the position relative to the Earth at the time of the event.

The ITRS frame is identified with WCSs that use the `TLON-` and `TLAT-` coordinate types. There are several examples of WCSs where this syntax is used to describe terrestrial coordinate systems: Section 7.4.1 of the WCS in FITS "Paper II", and the WCSTools docs: http://tdc-www.harvard.edu/software/wcstools/wcstools.multiwcs.html

This is the technique that I am using to make images such as this one: http://ligo.org/detections/GW170817/images-GW170817/skymap+earth.jpg

### CODE REVIEW

File: astropy/wcs/utils.py [hunk 1]
```diff
@@ -126,6 +128,11 @@ def _celestial_frame_to_wcs_builtin(frame, projection='TAN'):
     elif isinstance(frame, Galactic):
         xcoord = 'GLON'
         ycoord = 'GLAT'
+    elif isinstance(frame, ITRS):
+        xcoord = 'TLON'
+        ycoord = 'TLAT'
+        wcs.wcs.radesys = 'ITRS'
+        wcs.wcs.date_obs = frame.obstime
```
Comment (nden) [inline astropy/wcs/utils.py]: The attribute is `wcs.wcs.dateobs`. I think this will fail with `AttributeError`.

**Comment by nden** [comment]
The image is neat!

I left one inline comment. I wonder what woul dbe a test for this - perhaps roundtripping with a coordinates frame?

File: astropy/wcs/utils.py (hunk 1 above)
Comment (lpsinger) [reply astropy/wcs/utils.py]: Right you are! Fixed.

**Comment by lpsinger** [comment]
> I left one inline comment. I wonder what woul dbe a test for this - perhaps roundtripping with a coordinates frame?

Would an image example in the `astropy.visualization.wcsaxes` documentation count as a test?

**Comment by nden** [comment]
@lpsinger Regarding the test, I was thinking about purely converting between coordinates.ITRS and wcs and back to ITRS. For  example, looking at the code I'm not sure how this line will work

```
wcs.dateobs = itrs.obsdate
```

The WCS attribute expects a string and it is given the `Time` object. Should it be assigned `str(itrs.obstime)` instead?

**Comment by lpsinger** [comment]
> @lpsinger Regarding the test, I was thinking about purely converting between coordinates.ITRS and wcs and back to ITRS.

OK, I added unit tests.

> For example, looking at the code I'm not sure how this line will work
> 
> wcs.dateobs = itrs.obsdate
> The WCS attribute expects a string and it is given the Time object. Should it be assigned str(itrs.obstime) instead?

Good catch! Fixed.

**Comment by lpsinger** [comment]
The CI failures seem to be unrelated to this patch.

**Comment by nden** [comment]
@lpsinger Could you rebase again on astropy master. This will pick up some recent fixes to the astropy build and hopefully fix the tests errors (which are unrelated) but for some reason they don't go away. Sorry this is dragging for such a long time.
Let me know if you don't have time and I'll rebase it on my side.

**Comment by lpsinger** [comment]
> @lpsinger Could you rebase again on astropy master.

Done.

**Comment by lpsinger** [comment]
The CI failures still look unrelated.

**Comment by nden** [comment]
Yeah, this is the common failure now on master due to numpy 1.14. Thanks for rebasing.

**Comment by nden** [comment]
This needs one more rebase, sorry about that.

**Comment by lpsinger** [comment]
Looks like the tests are passing now! 🎉

**Comment by nden** [comment]
:tada: Thank you. Merging as coverage does not work and everything else is green.

**Pull request merged**